consensus/state.go

```diff
@@ -1012,19 +1306,17 @@ func (cs *ConsensusState) addVote(valIndex int, vote *types.Vote, peerKey string
 				}
 				if cs.Round <= vote.Round && prevotes.HasTwoThirdsAny() {
 					// Round-skip over to PrevoteWait or goto Precommit.
-					go func() {
-						cs.EnterNewRound(height, vote.Round, false)
-						if prevotes.HasTwoThirdsMajority() {
-							cs.EnterPrecommit(height, vote.Round, false)
-						} else {
-							cs.EnterPrevote(height, vote.Round, false)
-							cs.EnterPrevoteWait(height, vote.Round)
-						}
-					}()
+					cs.EnterNewRound(height, vote.Round) // if the vote is ahead of us
```

Comment:
Yup. Really I was hoping to do away with those checks in the Enter*, but I think it will require a lot more refactoring (but thats probably what the pi-calc impl will look more like).

Can we make the Enter* functions enter*, since they're now not protected by mtx and should only really be entered from handleTimeout, handleMsg, or eachother?